File: Problems/No.2401_Longest_Nice_Subarray/longest_nice_subarray.cpp
class Solution {
public:
    int longestNiceSubarray(vector<int>& nums) {
        int bit_used=0, lft=0, max_length=0;

        for (int rht=0; rht<nums.size(); rht++) {

            while ((bit_used & nums[rht]) != 0) {
                bit_used ^= nums[lft];
                lft++;
            }
            bit_used |= nums[rht];
            max_length = max(max_length, rht-lft+1);

        }

        return max_length;

    }
};

File: Problems/No.2401_Longest_Nice_Subarray/longest_nice_subarray.py
class Solution(object):
    def longestNiceSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        bit_used, lft_id, max_length = 0, 0, 0

        for rht_id, rht_num in enumerate(nums):
            while ((bit_used & rht_num) != 0):
                bit_used ^= nums[lft_id]
                lft_id += 1
            
            bit_used |= rht_num
            max_length = max(max_length, rht_id-lft_id+1)
        
        return max_length
